backend/shared/cosmo_db.py

```diff
@@ -11,32 +11,122 @@
 AZURE_DB_URI = f"https://{AZURE_DB_ID}.documents.azure.com:443/"
 
 def get_cosmos_container():
+    """
+    Establishes the connection to the Cosmos DB `reports` container.
+    """
+    credential = DefaultAzureCredential()
```

Comment:
@ramirezmorac2 This code is repeated in almost every function. Consider using get_cosmos_container() or a helper function to reuse the container connection. This will make the code more DRY (Don’t Repeat Yourself) and potentially improve performance by reusing connections.

```python
credential = DefaultAzureCredential()
client = CosmosClient(AZURE_DB_URI, credential, consistency_level="Session")
db = client.get_database_client(database=AZURE_DB_NAME)
container = db.get_container_client("reports")
```